### PR TITLE
Add default object for npm_dependencies

### PR DESCRIPTION
**Pull request by nagpalkaran95**
* Fix for: run_settings.specs wasn't working for windows when npm_dependencies was not specified

### CODE REVIEW

File: bin/commands/runs.js [hunk 1]
```diff
@@ -20,6 +20,9 @@ module.exports = function run(args) {
     // setting setDefaultAuthHash to {} if not present and set via env variables or via args.
     utils.setDefaultAuthHash(bsConfig,args);
 
+    // setting npm_dependencies to {} if not present
```
Comment (karanshah-browserstack) [inline bin/commands/runs.js:23]: Can we integrate both the defaults ?? And rename the function as setDefaults?

File: bin/commands/runs.js [hunk 2]
```diff
@@ -20,6 +20,9 @@ module.exports = function run(args) {
...
+    utils.setDefaultNpmHash(bsConfig);
```
Comment (karanshah-browserstack) [inline bin/commands/runs.js:24]: Also, this might break existing unit test cases too

**Review by karanshah-browserstack — CHANGES_REQUESTED** [review]

File: bin/commands/info.js [hunk 3]
```diff
@@ -10,9 +10,8 @@ module.exports = function info(args) {
   let bsConfigPath = utils.getConfigPath(args.cf);
 
   return utils.validateBstackJson(bsConfigPath).then(function (bsConfig) {
-    // setting setDefaultAuthHash to {} if not present and set via env variables or via args.
-    utils.setDefaultAuthHash(bsConfig, args);
-    
+    utils.setDefaults(bsConfig, args);
```
Comment (karanshah-browserstack) [inline bin/commands/info.js:13]: Add a comment ?

File: test/unit/bin/helpers/utils.js [hunk 4]
```diff
@@ -927,22 +927,22 @@ describe('utils', () => {
...
+      utils.setDefaults(bsConfig, {username: 'username'});
       expect(utils.isUndefined(bsConfig.auth)).to.be.false;
     });
 
-    it('should set setDefaultAuthHash if process.env.BROWSERSTACK_USERNAME is present and args.username is not present', () => {
+    it('should set setDefaults if process.env.BROWSERSTACK_USERNAME is present and args.username is not present', () => {
       let bsConfig = {};
       process.env.BROWSERSTACK_USERNAME = 'username';
-      utils.setDefaultAuthHash(bsConfig, {});
+      utils.setDefaults(bsConfig, {});
```
Comment (karanshah-browserstack) [inline test/unit/bin/helpers/utils.js:939]: No new test cases for npm_dependencies ?

**Review by karanshah-browserstack — DISMISSED** [review]
Rest LGTM

**Review by karanshah-browserstack — APPROVED** [review]

**Review by suryart — APPROVED** [review]

**Pull request merged**